### DISCUSSION TITLE
Function definition vector default parameter

### DISCUSSION THREAD

**Discussion by alxsimon** [MesserLab/SLiM#337] [Q&A] [answered]
Hi, more of an Eidos coding question.

I don't seem to be able to use a vector (of floats in this case) as an optional parameter with defaults.

I don't know if it's just not possible in Eidos or just me coding this wrongly.

e.g.:
```
function (void)myfunction(object<Dictionary>$ model, [float x_bounds=c(0.0, 1.0)]) {
    ...
}
```
My issue is with the second parameter `x_bounds`.
The error I get is this:
`ERROR (EidosScript::Match): unexpected token '(' in parameter specifier; expected ']'.`

Thanks!
Alexis

**Comment by bhaller** [comment, marked as answer]
Hi!  Yes, this is something you can do in R but not in Eidos.  Default values have to be simple constants; they cannot involve calculations.  So a call to `c()` is not allowed; even something like `5+5` is not allowed.  Typically in this situation I make `NULL` be the default value for the parameter, and then check in the function's script for a `NULL` value and replace it with the desired default.  Not great, but it works.

**Reply by alxsimon** [reply]
I see, thanks for the trick!
